File: javaSample/src/main/java/com/cleveradssolutions/sampleapp/appopen/AppOpenAdActivity.java
package com.cleveradssolutions.sampleapp.appopen;

import static com.cleveradssolutions.sampleapp.SampleApplication.CAS_ID;

import android.os.Bundle;

import androidx.annotation.NonNull;

import com.cleveradssolutions.sampleapp.R;
import com.cleveradssolutions.sampleapp.support.BackNavigationActivity;
import com.cleveradssolutions.sampleapp.support.Util;
import com.cleveradssolutions.sdk.AdContentInfo;
import com.cleveradssolutions.sdk.AdFormat;
import com.cleveradssolutions.sdk.screen.CASAppOpen;
import com.cleveradssolutions.sdk.screen.ScreenAdContentCallback;
import com.cleversolutions.ads.AdError;

public class AppOpenAdActivity extends BackNavigationActivity {

    ScreenAdContentCallback screenAdContentCallback = new ScreenAdContentCallback() {
        @Override
        public void onAdLoaded(@NonNull AdContentInfo ad) {
            Util.toast(AppOpenAdActivity.this, "App Open Ad loaded");
        }

        @Override
        public void onAdFailedToLoad(@NonNull AdFormat format, @NonNull AdError error) {
            Util.toastError(AppOpenAdActivity.this, "App Open Ad failed to load: " + error.getMessage());
        }

        @Override
        public void onAdFailedToShow(@NonNull AdFormat format, @NonNull AdError error) {
            Util.toastError(AppOpenAdActivity.this, "App Open Ad show failed: " + error.getMessage());
        }

        @Override
        public void onAdShowed(@NonNull AdContentInfo ad) {
            Util.toast(AppOpenAdActivity.this, "App Open Ad shown");
        }

        @Override
        public void onAdClicked(@NonNull AdContentInfo ad) {
            Util.toast(AppOpenAdActivity.this, "App Open Ad clicked");
        }

        @Override
        public void onAdDismissed(@NonNull AdContentInfo ad) {
            Util.toast(AppOpenAdActivity.this, "App Open Ad closed");
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.screen_ad_activity);
        setTitle(R.string.adFormats_appOpenAd);

        CASAppOpen appOpenAd = new CASAppOpen(CAS_ID);
        appOpenAd.setContentCallback(screenAdContentCallback);
        appOpenAd.setAutoloadEnabled(false); // by default
        appOpenAd.setAutoshowEnabled(false); // by default

        appOpenAd.setOnImpressionListener(ad ->
                Util.toast(this, "App Open Ad impression: " + ad.getRevenue())
        );

        findViewById(R.id.load).setOnClickListener(v -> appOpenAd.load(this));
        findViewById(R.id.show).setOnClickListener(v -> appOpenAd.show(this));
    }

}

File: kotlinSample/src/main/java/com/cleveradssolutions/sampleapp/appopen/AppOpenAdActivity.kt
package com.cleveradssolutions.sampleapp.appopen

import android.os.Bundle
import android.widget.Button
import com.cleveradssolutions.sampleapp.R
import com.cleveradssolutions.sampleapp.SampleApplication.Companion.CAS_ID
import com.cleveradssolutions.sampleapp.support.BackNavigationActivity
import com.cleveradssolutions.sampleapp.support.toast
import com.cleveradssolutions.sampleapp.support.toastError
import com.cleveradssolutions.sdk.AdContentInfo
import com.cleveradssolutions.sdk.AdFormat
import com.cleveradssolutions.sdk.OnAdImpressionListener
import com.cleveradssolutions.sdk.screen.CASAppOpen
import com.cleveradssolutions.sdk.screen.ScreenAdContentCallback
import com.cleversolutions.ads.AdError

class AppOpenAdActivity : BackNavigationActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.screen_ad_activity)
        setTitle(R.string.adFormats_appOpenAd)

        val appOpenAd = CASAppOpen(CAS_ID)
        appOpenAd.isAutoloadEnabled = false // by default
        appOpenAd.isAutoshowEnabled = false // by default

        appOpenAd.contentCallback = object : ScreenAdContentCallback() {
            override fun onAdLoaded(ad: AdContentInfo) {
                toast("App Open Ad loaded")
            }

            override fun onAdFailedToLoad(format: AdFormat, error: AdError) {
                toastError("App Open Ad failed to load: ${error.message}")
            }

            override fun onAdFailedToShow(format: AdFormat, error: AdError) {
                toastError("App Open Ad show failed: ${error.message}")
            }

            override fun onAdShowed(ad: AdContentInfo) {
                toast("App Open Ad shown")
            }

            override fun onAdClicked(ad: AdContentInfo) {
                toast("App Open Ad clicked")
            }

            override fun onAdDismissed(ad: AdContentInfo) {
                toast("App Open Ad closed")
            }
        }

        // Optional set Impression listener
        appOpenAd.onImpressionListener = OnAdImpressionListener { ad ->
            toast("App Open Ad impression: ${ad.revenue}")
        }

        findViewById<Button>(R.id.load).setOnClickListener {
            appOpenAd.load(this)
        }

        findViewById<Button>(R.id.show).setOnClickListener {
            appOpenAd.show(this)
        }
    }

}